pallets/creators-staking/src/types.rs

```diff
@@ -0,0 +1,417 @@
+    /// Information about unbonding chunks.
+    pub(super) unbonding_info: UnbondingInfo<Balance>,
+}
+
+impl<Balance: AtLeast32BitUnsigned + Default + Copy + MaxEncodedLen> AccountLedger<Balance> {
+    /// `true` if ledger is empty (no locked funds, no unbonding chunks), `false` otherwise.
+    pub fn is_empty(&self) -> bool {
+        self.locked.is_zero() && self.unbonding_info.is_empty()
+    }
+}
+
+/// Used to represent how much was staked in a particular era.
+/// E.g. `{staked: 1000, era: 5}` means that in era `5`, staked amount was 1000.
+#[derive(Encode, Decode, Clone, Copy, PartialEq, Eq, RuntimeDebug, TypeInfo, MaxEncodedLen)]
+pub struct EraStake<Balance: AtLeast32BitUnsigned + Copy + MaxEncodedLen> {
+    /// Staked amount in era
+    #[codec(compact)]
+    staked: Balance,
+    /// Staked era
+    #[codec(compact)]
+    era: EraIndex,
+}
+
+impl<Balance: AtLeast32BitUnsigned + Copy + MaxEncodedLen> EraStake<Balance> {
+    /// Create a new instance of `EraStake` with given values
+    fn new(staked: Balance, era: EraIndex) -> Self {
+        Self { staked, era }
+    }
+}
+
+/// Used to provide a compact and bounded storage for information about stakes in unclaimed eras.
+///
+/// In order to avoid creating a separate storage entry for each `(staker, creator, era)` triplet,
+/// this struct is used to provide a more memory efficient solution.
+///
+/// Basic idea is to store `EraStake` structs into a vector from which a complete
+/// picture of **unclaimed eras** and stakes can be constructed.
+///
+/// # Example
+/// For simplicity, the following example will represent `EraStake` using `<era, stake>` notation.
+/// Let us assume we have the following vector in `StakerInfo` struct.
+///
+/// `[<5, 1000>, <6, 1500>, <8, 2100>, <9, 0>, <11, 500>]`
+///
+/// This tells us which eras are unclaimed and how much it was staked in each era.
+/// The interpretation is the following:
+/// 1. In era **5**, staked amount was **1000** (interpreted from `<5, 1000>`)
+/// 2. In era **6**, staker staked additional **500**, increasing total staked amount to **1500**
+/// 3. No entry for era **7** exists which means there were no changes from the former entry.
+///    This means that in era **7**, staked amount was also **1500**
+/// 4. In era **8**, staker staked an additional **600**, increasing total stake to **2100**
+/// 5. In era **9**, staker unstaked everything from the creator (interpreted from `<9, 0>`)
+/// 6. No changes were made in era **10** so we can interpret this same as the previous entry which
+/// means **0** staked amount. 7. In era **11**, staker staked **500** on the creator, making his
+/// stake active again after 2 eras of inactivity.
+///
+/// **NOTE:** It is important to understand that staker **DID NOT** claim any rewards during this
+/// period.
+#[derive(Encode, Decode, Clone, Default, PartialEq, Eq, RuntimeDebug, TypeInfo)]
+pub struct StakerInfo<Balance: AtLeast32BitUnsigned + Copy + MaxEncodedLen> {
```

Comment:
To implement `MaxEncodedLen` for it in order to keep `StorageInfo` (storage metadata) for the pallet.
Everything is documented well above. Another way of doing it is to use `BoundedVec`, which implements `MaxEncodedLen` by default, but it will add more restrictions, which we haven't thought of fixing it.